python_modules/dagster/dagster/_cli/run.py

```diff
@@ -70,3 +71,70 @@ def run_wipe_command(force):
         click.echo("Deleted all run history and event logs.")
     else:
         raise click.ClickException("Exiting without deleting all run history and event logs.")
+
+
+@run_cli.command(name="migrate", help="Migrate the run history from one repository to another.")
+@click.option(
+    "--from",
+    "-f",
+    "from_label",
+    help="The repository from which to migrate (format: <repository_name>@<location_name>)",
+    prompt_required=True,
+)
+@click.option(
+    "--to",
+    "-t",
+    "to_label",
+    help="The repository name to migrate to (format: <repository_name>@<location_name>)",
+    prompt_required=True,
+)
+@click.argument("job_name")
+def run_migrate_command(job_name, from_label, to_label):
+    from dagster._core.storage.pipeline_run import RunsFilter
+    from dagster._core.storage.runs.schema import RunTagsTable
+    from dagster._core.storage.runs.sql_run_storage import SqlRunStorage
+    from dagster._core.storage.tags import REPOSITORY_LABEL_TAG
+
+    if not from_label or not to_label:
+        raise click.UsageError("Please specify both a --from and --to repository label")
+
+    if not is_valid_repo_label(from_label) or not is_valid_repo_label(to_label):
+        raise click.UsageError(
+            "`--from` and `--to` arguments must be of the format: <repository_name>@<location_name>"
+        )
+
+    with DagsterInstance.get() as instance:
+        records = instance.get_run_records(
+            filters=RunsFilter(job_name=job_name, tags={REPOSITORY_LABEL_TAG: from_label})
+        )
+
+        if not records:
+            click.echo(f"No runs found for {job_name} in {from_label}.")
+            return
+
+        if not isinstance(instance.run_storage, SqlRunStorage):
+            return
+
+        count = len(records)
+        confirmation = click.prompt(
+            f"Are you sure you want to migrate the run history for {job_name} ({count} runs)? Type MIGRATE"
+        )
+        should_migrate = confirmation == "MIGRATE"
+
+        if should_migrate:
+            for record in tqdm(records):
+                with instance.run_storage.connect() as conn:
```

Comment:
keep connection through loop - no need to reconnect each iteration

Reply:
little spicy depending on `run_storage.connect` here like this but i suppose the test helps protect (at least the one impl that the test uses)